module/VuFind/tests/unit-tests/src/VuFindTest/ILS/Logic/HoldsTest.php

```diff
@@ -98,4 +101,322 @@ public function testGetSuppressedLocations(array $configArray, array $expectedLi
         $logic = $this->getHoldsLogic(config: $configArray);
         $this->assertEquals($expectedList, $logic->getSuppressedLocations());
     }
+
+    /**
+     * Test formatHoldings method
+     *
+     * @return void
+     */
+    public function testFormatHoldings(): void
+    {
+        $catalog = $this->createMock(Connection::class);
+        $catalog->method('getHoldingsTextFieldNames')->willReturn(['notes', 'summary']);
+
+        $logic = $this->getHoldsLogic(catalog: $catalog);
+
+        // Use reflection to access the protected formatHoldings method
+        $reflection = new \ReflectionClass($logic);
+        $method = $reflection->getMethod('formatHoldings');
+        $method->setAccessible(true);
+
+        // Example holdings data
+        // note that this is a simplified example; actual data may vary and include an availability object
+        $holdings = $this->getJsonFixture('ils/holdings_example.json');
+
+        $result = $method->invoke($logic, $holdings);
```

Comment:
I think you could use the `callMethod` convenience method in [ReflectionTrait](https://github.com/vufind-org/vufind/blob/dev/module/VuFind/src/VuFindTest/Feature/ReflectionTrait.php) to eliminate the need for local reflection logic inside the test.

Reply:
Good point, I tried to address it with the latest commit